### PR TITLE
Add FC084: Deprecated Chef::REST class used

### PR DESCRIPTION
**Pull request by tas50**
Signed-off-by: Tim Smith <tsmith@chef.io>

### CODE REVIEW

File: lib/foodcritic/rules/fc084.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+rule "FC084", "Deprecated Chef::REST class used" do
+  tags %w{chef13 deprecated}
+  def rest(ast)
+    ast.xpath('//const_path_ref/const[@value="REST"]/..//const[@value="Chef"]/../../..')
+  end
+  recipe { |ast| rest(ast) }
+  library { |ast| rest(ast) }
+
```
Comment (coderanger) [inline lib/foodcritic/rules/fc084.rb]: Robucop will probably be sad about this blank line.

File: lib/foodcritic/rules/fc084.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+rule "FC084", "Deprecated Chef::REST class used" do
+  tags %w{chef13 deprecated}
+  def rest(ast)
+    ast.xpath('//const_path_ref/const[@value="REST"]/..//const[@value="Chef"]/../../..')
```
Comment (coderanger) [inline lib/foodcritic/rules/fc084.rb]: Might be easier to say something like this `//const_path_ref[const[@value="Chef"]][const[@value="REST"]]`? Not sure that's exactly equivalent though.

File: lib/foodcritic/rules/fc084.rb (hunk 1 above)
Comment (tas50) [inline lib/foodcritic/rules/fc084.rb]: loves it clearly

File: lib/foodcritic/rules/fc084.rb (hunk 2 above)
Comment (tas50) [inline lib/foodcritic/rules/fc084.rb]: That doesn't detect it

**Pull request merged**